pkg/controller/deployment/deployment_controller.go

```diff
@@ -450,6 +451,11 @@ func (dc *DeploymentController) syncRollingUpdateDeployment(deployment extension
 		return dc.updateDeploymentStatus(allRCs, newRC, deployment)
 	}
 
+	// Cleanup old rcs
+	if deployment.Spec.RevisionHistoryLimit != nil {
```

Comment:
Let's have the cleanup policy separated from the strategy code. How about moving this after the switch that picks a strategy?